packages/react/src/components/Table/TableBody/TableBodyRowRenderer.jsx

```diff
@@ -192,21 +191,18 @@ const TableBodyRowRenderer = (props) => {
 
   if (shouldLazyRender && !isVisible) {
     return (
-      <tr
-        key={`lazy-row-${row.id}`}
-        ref={rowVisibilityRef}
-        data-testid={`${tableId}-lazy-row-${row.id}`}
```

Comment:
@Shady-Nessim This is the source of your `[data-testid="/lazy-row/i"]` failure. By replacing this with the new `SkeletonRow` component, it uses a different `data-testid`, so maintain consistency, you could change the `testId` prop on the the SkeletonRow component so that it overrides the testId entirely instead of inserting it into the string. So by default the SkeletonRow's testId can be simply[`${tableId}-${testId}`](https://github.com/carbon-design-system/carbon-addons-iot-react/pull/3239/files#diff-cf9d26532604dbcc58c7c16672f1f12aae701eb962096455f325bf5d79e26fe9R56) with a default `testId` of `skeleton`. But with the shouldLazyRender check we could pass 
```testId={`${tableId}-lazy-row-${row.id}`}``` instead.